releng_sop/koji_clone_tag_for_release_milestone.py

```diff
@@ -0,0 +1,143 @@
+
+from .common import Environment, Release
+
+
+class KojiCloneTagForReleaseMilestone(object):
+    """Clone tag for release milestone."""
+
+    def __init__(self, env, release_id, milestone):
+        """Init.
+
+        Patameters
+        ----------
+        env: str
+            name of the environment to be used to execute the commands
+        release_id: str
+        milestone: str
+
+        """
+        self.env = env
+        self.release_id = release_id
+        self.release = Release(self.release_id)
+        self.milestone = milestone
+        self.compose_tag = self.release["koji"]["tag_compose"]
+        self.milestone_tag = self._get_milestone_tag(milestone)
+
+    def _get_milestone_tag(self, milestone):
+        verify_milestone(milestone)
+        result = "%s-%s-set" % (self.release["koji"]["tag_release"], self.milestone.lower().split(".")[0])
+        return result
+
+    def print_details(self, commit=False):
+        """Print details of command execution.
+
+        Parameters
+        ----------
+        commit: boolean (optional; default False)
+            Flag to indicate if the command will be actually executed.
+            Line indicating "test mode" is printed, if this is False.
+        """
+        print("Cloning package set for a release milestone")
+        print(" * koji profile:            %s" % self.env["koji_profile"])
+        print(" * release_id:              %s" % self.release_id)
+        print(" * milestone:               %s" % self.milestone)
+        print(" * compose tag (source):    %s" % self.compose_tag)
+        print(" * milestone tag (target):  %s" % self.milestone_tag)
+        if not commit:
+            print("*** TEST MODE ***")
+
+    def get_cmd(self, commit=False):
+        """Construct the koji command.
+
+        Parameters
+        ----------
+        commit: boolean (optional; default False)
+            Flag to indicate if the command will be actually executed.
+            Add "--test" to the command, if this is False.
+
+        Returns
+        -------
+        koji command as a list of strings.
+        """
+        cmd = []
+        cmd.append("koji")
+        cmd.append("--profile=%s" % self.env["koji_profile"])
+        cmd.append("clone-tag")
+        cmd.append("--verbose")
+        cmd.append(self.compose_tag)
+        cmd.append(self.milestone_tag)
+        if not commit:
+            cmd.append('--test')
+        return cmd
+
+    def run(self, commit=False):
+        """Print command details, get command and run it."""
+        self.print_details(commit=commit)
+        cmd = self.get_cmd(commit=commit)
+        print(cmd)
+        subprocess.check_output(cmd)
+
+
+def get_parser():
+    """Construct argument parser.
+
+    Returns
+    -------
+    ArgumentParser object with arguments set up.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "release_id",
+        metavar="RELEASE_ID",
+        help="",

```

Comment:
Help text would be nice.
For common parameters between the commands a common place would be nice to define all the strings (and avoid copy-pasting, or forgetting them)....
For now copy-paste from block-package would be good enough, I think.